packages/replay-internal/src/util/createPerformanceEntries.ts

```diff
@@ -225,13 +229,24 @@ export function getInteractionToNextPaint(metric: Metric): ReplayPerformanceEntr
 export function getWebVital(
   metric: Metric,
   name: string,
-  node: Node | undefined,
+  node: Node | Node[] | undefined,
 ): ReplayPerformanceEntry<WebVitalData> {
   const value = metric.value;
   const rating = metric.rating;
 
   const end = getAbsoluteTime(value);
 
+  const nodeIds: number[] = [];
+  if (Array.isArray(node)) {
```

Comment:
This might be easier to follow and less code:

```suggestion
  const nodes = Array.isArray(node) ? node : [];
  const nodeIds = nodes.map(node => record.mirror.getId(node);
```